### ISSUE TITLE
FlexibleDataManagerConfig defaults forces training on a single image

### ISSUE THREAD

**Issue by Jamy-L** [hummat/sdfstudio#18]
### Description

 There seems to be a problematic default interaction in the dataloader config:

  - `FlexibleDataManagerConfig` sets `train_num_images_to_sample_from = 1` (forced by code design)
  - `train_num_times_to_repeat_images` is inherited from
  `VanillaDataManagerConfig` and stays at `-1`

  This means that any method using `FlexibleDataManagerConfig` with its native defaults will cache one training image once and never resample another one. In other words, geo-neus and similar methods will by default overfit a single image. The training loop ends up doing:

  - pick one training image at startup
  - sample different rays from that same image forever
  - never rotate to other training images

  ## Where this comes from

  Relevant defaults:

  - `VanillaDataManagerConfig.train_num_images_to_sample_from = -1`
  - `VanillaDataManagerConfig.train_num_times_to_repeat_images = -1`
  - `FlexibleDataManagerConfig.train_num_images_to_sample_from = 1`

  ## Suggested fix

  1. set FlexibleDataManagerConfig.train_num_times_to_repeat_images = 0
  2. add a guard/warning when train_num_images_to_sample_from is small and
     train_num_times_to_repeat_images = -1

  because the current inherited default is very surprising and easy to miss.

### Steps to Reproduce

Train geo-neus, disable the warp-loss to get what should be a regular neus training, and observe the result overfitting a single view. The train loss will be low, val loss very high.

### Command Used

```bash
sdf-train geo-neus --pipeline.model.sdf-field.inside-outside False --pipeline.model.patch-warp-loss-mult 0.0 --pipeline.datamanager.train-num-times-to-repeat-images 0 --vis tensorboard --experiment-name geo-neus-dtu65 sdfstudio-data --data data/sdfstudio-demo-data/dtu-scan65/ --load-pairs True
```

### Method

Other

### Feature Area

Training (sdf-train)

### Python Version

3.10

### GPU

a100

### Operating System

Redhat

### Error Logs

```shell

```

### Checklist

- [x] I have searched existing issues to ensure this bug hasn't been reported

**Comment by hummat** [comment]
Thanks for the report, I'll look into it asap.
